modyn/supervisor/internal/supervisor.py

```diff
@@ -0,0 +1,368 @@
+            )
+
+        incremental_model_strategy: Optional[ModelStorageStrategyConfig] = None
+        if incremental_model_strategy_config is not None:
+            incremental_model_strategy = ModelStorageStrategyConfig.from_config(incremental_model_strategy_config)
+
+        with self._next_pipeline_lock:
+            with MetadataDatabaseConnection(self.modyn_config) as database:
+                pipeline_id = database.register_pipeline(
+                    num_workers=num_workers,
+                    model_class_name=pipeline_config["model"]["id"],
+                    model_config=model_config,
+                    amp=pipeline_config["training"]["amp"] if "amp" in pipeline_config["training"] else False,
+                    selection_strategy=json.dumps(pipeline_config["training"]["selection_strategy"]),
+                    full_model_strategy=full_model_strategy,
+                    incremental_model_strategy=incremental_model_strategy,
+                    full_model_interval=full_model_interval,
+                )
+
+        return pipeline_id
+
+    @staticmethod
+    def get_model_strategy(strategy_config: dict) -> StrategyConfig:
+        return StrategyConfig(
+            name=strategy_config["name"],
+            zip=strategy_config["zip"] if "zip" in strategy_config else None,
+            zip_algorithm=strategy_config["zip_algorithm"] if "zip_algorithm" in strategy_config else None,
+            config=SelectorJsonString(value=json.dumps(strategy_config["config"]))
+            if "config" in strategy_config
+            else None,
+        )
+
+    def unregister_pipeline(self, pipeline_id: int) -> None:
+        # TODO(#64,#124,#317): Implement.
+        pass
+
+    def start_pipeline(
+        self,
+        pipeline_config: dict,
+        eval_directory: str,
+        start_replay_at: Optional[int] = None,
+        stop_replay_at: Optional[int] = None,
+        maximum_triggers: Optional[int] = None,
+    ) -> int:
+        if not self.validate_pipeline_config(pipeline_config):
+            raise ValueError("Invalid pipeline configuration")
+
+        if not is_directory_writable(pathlib.Path(eval_directory)):
+            raise ValueError("No permission to write to the evaluation results directory.")
+
+        if not self.validate_system(pipeline_config):
+            raise ValueError("Invalid system configuration")
+
+        exception_queue: mp.Queue[str] = mp.Queue()  # pylint: disable=unsubscriptable-object
+        status_query_queue: mp.Queue[str] = mp.Queue()  # pylint: disable=unsubscriptable-object
+        status_response_queue: mp.Queue[dict[str, Any]] = mp.Queue()  # pylint: disable=unsubscriptable-object
+
+        start_timestamp = self.grpc.get_time_at_storage()
+        pipeline_id = self.register_pipeline(pipeline_config)
+        logger.info(f"Pipeline {pipeline_id} registered, start executing.")
+
+        logger.info(f"start_pipeline: start method is {mp.get_start_method()}")
```

Comment:
Debug leftovers?

Reply:
Yes, will remove it when the PR is ready to merge.

Reply:
done